apiserver/root.go

```diff
@@ -148,18 +148,23 @@ func newAPIHandler(
 	connectionID uint64,
 	serverHost string,
 ) (*apiHandler, error) {
-	m, err := st.Model()
+	// We return a core errors NotFound because we are still mixing
+	// mongo State and domain service calls and this is easier till
+	// we fully convert across.
+	exists, err := domainServices.Model().CheckModelExists(ctx, modelUUID)
 	if err != nil {
-		if !errors.Is(err, errors.NotFound) {
-			return nil, errors.Trace(err)
+		if errors.Is(err, modelerrors.NotFound) {
```

Comment:
You won't get a not found error out of this func anymore. That is just a false value.

Reply:
Missed that when I made the change, thanks.